### PR TITLE
[OpenCL] Add Transform passes which support SYCL and OCL

### PR DESCRIPTION
**Pull request by cdai2**
LIT test cases for them are also included.

### CODE REVIEW

**Comment by cdai2** [comment]
This  PR is part of #8216.  It's created for better code review.

File: clang/tools/driver/CMakeLists.txt
Comment (bader) [inline clang/tools/driver/CMakeLists.txt:17]: Are there any changes in FE compiler/driver source code? If so, there should be no need to link this library.

Comment (hewj03) [reply clang/tools/driver/CMakeLists.txt:17]: this is legacy code from high-perf host device

**Comment by cdai2** [comment]
> @cdai2, please, fix pre-commit checks.

updated.

**Review by bader — APPROVED** [review]
LGTM.

@cdai2, we code owners approval to unblock the merge. Due to the bug in CODEOWNERS file, I mentioned [here](https://github.com/intel/llvm/pull/8241#issuecomment-1423070157) this code must be approved by the tools team. Please, fix CODEOWNERS file ASAP to get reviews for OpenCL patches in time.

**Review by hewj03 — APPROVED** [review]
LGTM

**Comment by cdai2** [comment]
ping @intel/dpcpp-tools-reviewers.  Please review.

**Comment by bader** [comment]
[hewj03](https://github.com/hewj03) approval should be enough to merge.

**Comment by bader** [comment]
@cdai2, I've noticed a comment from @jbrodman related to this patch: https://github.com/intel/llvm/pull/8216#issuecomment-1424378284. Do you plan to make additional changes?

**Comment by cdai2** [comment]
> @cdai2, I've noticed a comment from @jbrodman related to this patch: [#8216 (comment)](https://github.com/intel/llvm/pull/8216#issuecomment-1424378284). Do you plan to make additional changes?

We preferred to use SYCL.  I replied in #8216.

**Comment by cdai2** [comment]
> @cdai2, I've noticed a comment from @jbrodman related to this patch: [#8216 (comment)](https://github.com/intel/llvm/pull/8216#issuecomment-1424378284). Do you plan to make additional changes?

@bader  The latest patch changed the folder name from SYCLTransforms to SPMDTransforms.

File: llvm/lib/Passes/PassBuilder.cpp
Comment (Nuullll) [inline llvm/lib/Passes/PassBuilder.cpp:234]: This line needs to be moved before `#include "llvm/Transforms/Scalar/ADCE.h"` -- clang-format would put **Capitalized** entries before **Uncapitalized** ones.

File: llvm/lib/Passes/PassBuilderPipelines.cpp
Comment (Nuullll) [inline llvm/lib/Passes/PassBuilderPipelines.cpp:118]: Same here.

File: llvm/lib/Passes/PassBuilder.cpp
Comment (cdai2) [reply llvm/lib/Passes/PassBuilder.cpp:234]: ok, I will update it.  I did clang-format before submit. It's strange.

**Review by Nuullll — APPROVED** [review]

**Pull request closed without merging**
